xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -455,16 +442,51 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
         if (!ch)
           continue;
 
-        textWidth += ch->m_advance;
+        const float nextWidth = textWidth + ch->m_advance;
 
-        if (textWidth > maxPixelWidth)
+        if (nextWidth > maxPixelWidth)
         {
           // Start rendering from the glyph that does not exceed the maximum width
           startPosGlyph = std::distance(itRGlyph, glyphs.crend());
           break;
         }
+        textWidth = nextWidth;
       }
     }
+    else
+    {
+      // Determines the maximum width the text can have within the maximum width
```

Comment:
not very clear, defines the max width with the max width?

Reply:
"Calculates the maximum width based on the characters that can be contained within the maximum width"
better?

Reply:
replaced